lib/ansible/modules/windows/win_certificate_store.ps1

```diff
@@ -3,35 +3,33 @@
 # Copyright: (c) 2017, Ansible Project
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-#Requires -Module Ansible.ModuleUtils.Legacy
+#Requires -Module Ansible.ModuleUtils.Basic
 
-$ErrorActionPreference = "Stop"
+$store_name_values = ([System.Security.Cryptography.X509Certificates.StoreName]).GetEnumValues() | ForEach-Object { $_.ToString() }
+$store_location_values = ([System.Security.Cryptography.X509Certificates.StoreLocation]).GetEnumValues() | ForEach-Object { $_.ToString() }
 
-$store_name_values = ([System.Security.Cryptography.X509Certificates.StoreName]).GetEnumValues()
-$store_location_values = ([System.Security.Cryptography.X509Certificates.StoreLocation]).GetEnumValues()
-
-$params = Parse-Args $args -supports_check_mode $true
-$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
-
-$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "absent", "exported", "present"
-$path = Get-AnsibleParam -obj $params -name "path" -type "path" -failifempty ($state -eq "present" -or $state -eq "exported")
-$thumbprint = Get-AnsibleParam -obj $params -name "thumbprint" -type "str" -failifempty ($state -eq "exported")
-$store_name = Get-AnsibleParam -obj $params -name "store_name" -type "str" -default "My" -validateset $store_name_values
-$store_location = Get-AnsibleParam -obj $params -name "store_location" -type "str" -default "LocalMachine" -validateset $store_location_values
-$password = Get-AnsibleParam -obj $params -name "password" -type "str"
-$key_exportable = Get-AnsibleParam -obj $params -name "key_exportable" -type "bool" -default $true
-$key_storage = Get-AnsibleParam -obj $params -name "key_storage" -type "str" -default "default" -validateset "default", "machine", "user"
-$file_type = Get-AnsibleParam -obj $params -name "file_type" -type "str" -default "der" -validateset "der", "pem", "pkcs12"
-
-$result = @{
-    changed = $false
-    thumbprints = @()
+$argument_spec = @{
+    state = @{ type = "str"; default = "present"; choices = "absent", "exported", "present" }
+    path = @{ type = "path" }
+    thumbprint = @{ type = "str" }
+    store_name = @{ type = "str"; default = "My"; choices = $store_name_values }
+    store_location = @{ type = "str"; default = "LocalMachine"; choices = $store_location_values }
+    password = @{ type = "str"; no_log = $true }
+    key_exportable = @{ type = "bool"; default = $true }
+    key_storage = @{ type = "str"; default = "default"; choices = "default", "machine", "user" }
```

Comment:
Just a minor nitpick.
Do we want to use implicit comma-separated list syntax, or prefer explicit array syntax instead ?

I always get confused by the comma-separated syntax, but I am not sure what is standard practice in the PowerShell world.

Reply:
One reason for using the more explicit array definition is that it can be nested.
(*The* reason for me to prefer the `$( ... )` syntax over using backticks in shell)